ledger/participant-integration-api/src/main/scala/platform/apiserver/error/PackageServiceError.scala

```diff
@@ -0,0 +1,277 @@
+        extends ErrorCode(id = "INVALID_ZIP_ENTRY", ErrorCategory.InvalidIndependentOfSystemState) {
+      final case class Error(name: String, entries: Seq[String])(implicit
+          val logger: ContextualizedLogger,
+          val loggingContext: LoggingContext,
+          val correlationId: CorrelationId,
+      ) extends BaseError.Impl(
+            cause = "Dar zip file is corrupt",
+            correlationId = correlationId.id,
+          )
+          with PackageServiceError
+    }
+
+    @Explanation(
+      """This error indicates that the supplied zipped dar is an unsupported legacy Dar."""
+    )
+    @Resolution("Please use a more recent dar version.")
+    object InvalidLegacyDar
+        extends ErrorCode(
+          id = "INVALID_LEGACY_DAR",
+          ErrorCategory.InvalidIndependentOfSystemState,
+        ) {
+      final case class Error(entries: Seq[String])(implicit
+          val logger: ContextualizedLogger,
+          val loggingContext: LoggingContext,
+          val correlationId: CorrelationId,
+      ) extends BaseError.Impl(
+            cause = "Unsupported legacy Dar zip file",
+            correlationId = correlationId.id,
+          )
+          with PackageServiceError
+    }
+
+    @Explanation("""This error indicates that the supplied zipped dar is regarded as zip-bomb.""")
+    @Resolution("Inspect the dar and contact support.")
+    object ZipBomb
+        extends ErrorCode(id = "ZIP_BOMB", ErrorCategory.InvalidIndependentOfSystemState) {
+      final case class Error(msg: String)(implicit
+          val logger: ContextualizedLogger,
+          val loggingContext: LoggingContext,
+          val correlationId: CorrelationId,
+      ) extends BaseError.Impl(
+            cause = "Dar zip file seems to be a zip bomb.",
+            correlationId = correlationId.id,
+          )
+          with PackageServiceError
+    }
+
+    @Explanation(
+      """This error indicates that the content of the Dar file could not be parsed successfully."""
+    )
+    @Resolution("Inspect the error message and contact support.")
+    object ParseError
+        extends ErrorCode(id = "DAR_PARSE_ERROR", ErrorCategory.InvalidIndependentOfSystemState) {
+      final case class Error(reason: String)(implicit
+          val logger: ContextualizedLogger,
+          val loggingContext: LoggingContext,
+          val correlationId: CorrelationId,
+      ) extends BaseError.Impl(
+            cause = "Failed to parse the dar file content.",
+            correlationId = correlationId.id,
+          )
+          with PackageServiceError
+    }
+
+  }
+
+  @Explanation("""This error indicates an internal issue within the package service.""")
+  @Resolution("Inspect the error message and contact support.")
+  object InternalError
+      extends ErrorCode(
+        id = "PACKAGE_SERVICE_INTERNAL_ERROR",
+        ErrorCategory.SystemInternalAssumptionViolated,
+      ) {
+    final case class DamleNotSet()(implicit
```

Comment:
this is very canton specific. you generally might want to avoid importing the actual error classes (i.e. the `DamlNotSet` and just introduce the error code (the static object)

Reply:
Removed this one. With regard to your 2nd statement, I was actually thinking of leveraging the existing error codes -> errors definitions as much as possible in order to increase reuse. Would you advise otherwise?